app/javascript/components/editor/TestRunSummaryContainer.tsx

```diff
@@ -0,0 +1,146 @@
+import React, { useEffect, useCallback, useRef } from 'react'
+import { TestRun, TestRunStatus } from './types'
+import { TestRunChannel } from '../../channels/testRunChannel'
+import { fetchJSON } from '../../utils/fetch-json'
+import { useRequestQuery } from '../../hooks/request-query'
+import { useIsMounted } from 'use-is-mounted'
+import { TestRunSummary } from './TestRunSummary'
+
+const REFETCH_INTERVAL = 2000
+
+export const TestRunSummaryContainer = ({
```

Comment:
I like how the callbacks are now no longer in the same file as most of the markup 👍